src/internet_identity/src/assets.rs

```diff
@@ -86,61 +87,95 @@ pub fn init_assets() {
-        ),
-    ]
+static ASSET_DIR: Dir<'_> = include_dir!("$CARGO_MANIFEST_DIR/../../dist");
+
+// Gets the static assets. All static assets are prepared only once (like injecting the canister ID).
+fn get_static_assets() -> Vec<(String, Vec<u8>, ContentEncoding, ContentType)> {
+    let mut assets = collect_assets_recursive(&ASSET_DIR);
+
+    // Required to make II available on the identity.internetcomputer.org domain.
+    // See https://internetcomputer.org/docs/current/developer-docs/production/custom-domain/#custom-domains-on-the-boundary-nodes
+    assets.push((
+        "/.well-known/ic-domains".to_string(),
+        b"identity.internetcomputer.org".to_vec(),
+        ContentEncoding::Identity,
+        ContentType::OCTETSTREAM,
+    ));
+
+    assets
+}
+
+fn collect_assets_recursive(dir: &Dir) -> Vec<(String, Vec<u8>, ContentEncoding, ContentType)> {
+    let mut assets = collect_assets_from_dir(dir);
+    for subdir in dir.dirs() {
+        assets.extend(collect_assets_recursive(subdir).into_iter());
+    }
+    assets
+}
+
+fn collect_assets_from_dir(dir: &Dir) -> Vec<(String, Vec<u8>, ContentEncoding, ContentType)> {
+    let mut assets: Vec<(String, Vec<u8>, ContentEncoding, ContentType)> = vec![];
+    for asset in dir.files() {
+        let file_bytes = asset.contents().to_vec();
+        let file_path = "/".to_string() + asset.path().to_str().unwrap();
+        let (asset_path, content, encoding, content_type) = match asset
+            .path()
+            .file_name()
+            .unwrap()
+            .to_str()
+            .unwrap()
+            .split_once('.')
+            .unwrap()
+            .1
+        {
+            "css" => (
+                file_path,
+                file_bytes,
+                ContentEncoding::Identity,
+                ContentType::CSS,
+            ),
+            "html" => (
+                file_path,
+                fixup_html(String::from_utf8_lossy(&file_bytes).as_ref())
+                    .as_bytes()
+                    .to_vec(),
+                ContentEncoding::Identity,
+                ContentType::HTML,
+            ),
+            "ico" => (
+                file_path,
+                file_bytes,
+                ContentEncoding::Identity,
+                ContentType::ICO,
+            ),
+            "js.gz" => (
+                file_path.chars().take(file_path.len() - 3).collect(), // drop ".gz"
+                file_bytes,
+                ContentEncoding::GZip,
+                ContentType::JS,
+            ),
+            "png" => (
+                file_path,
+                file_bytes,
+                ContentEncoding::Identity,
+                ContentType::PNG,
+            ),
+            "webp" => (
+                file_path,
+                file_bytes,
+                ContentEncoding::Identity,
+                ContentType::WEBP,
+            ),
+            _ => panic!("Unknown asset type: {}", asset.path().display()),
```

Comment:
I think we also need `webp` (most of the dapp icons) and `svg`

Reply:
`webp` is already supported. I'm happy to add support for more file types, but I would prefer to do this in a separate PR.